### DISCUSSION TITLE
Resize an image

### DISCUSSION THREAD

**Discussion by jehoshua7** [razonyang/hugo-theme-bootstrap#212] [General]
I have been using the following markdown in another theme to re-size an image

`![Tux, the Linux mascot](/images/tux.png?width=20pc "Photo by tux")`

Is there another method in this theme to do an image resize ?  Also, the comment/title of "Photo by tux" doesn't appear when the mouse is hovered (see https://www.digitalocean.com/community/tutorials/markdown-markdown-images for an example)

**Comment by razonyang** [comment]
This theme doesn't provide any tricks for resizing images so far.

**Comment by jehoshua7** [comment]
@razonyang - thanks. I have added a question at https://discourse.gohugo.io/t/resize-an-image-with-markdown/34833

**Reply by razonyang** [reply]
I'm sure that it is a feature of learn theme. If you take a close look at the source code(CTRL+U), you'll find that the image has not been resized yet, images were processed by JS.

But I think that is a good feature to have.

**Comment by jehoshua7** [comment]
@razonyang - Thanks, I have just seen https://github.com/razonyang/hugo-theme-bootstrap/issues/216

**Comment by teppos** [comment]
Oh cool, I was just searching for this feature in this theme. :+1:

**Comment by razonyang** [comment]
FYI @teppos. This feature was implemented by #216.

**Reply by teppos** [reply]
Nice! Have to try it today.

**Reply by teppos** [reply]
Hi,

Just tested it and seems to work fine. Good work! 👍 

Though I noticed the image is just styled to look smaller on the page. 
If you use (many) large images on the page it could potentially affect page loading time in the end.

But you could resize it using Hugos [Image processing](https://gohugo.io/content-management/image-processing/) to get smaller image sizes instead.

I experimented in **themes\hugo-theme-bootstrap\layouts\partials\helpers\image.html** with it and got it to work, see [gist](https://gist.github.com/teppos/bd0995fafe893a75e80ac64a9fc2d50d) 

When an image is not found as pageResource it will not resize it just style it as earlier.
As I mentioned, I just experimented with it. maybe there is a better/cleaner way to do it too.

**Reply by razonyang** [reply]
Yep, currently, it just adds the inline style to resize images.
Are you willing to submit a PR for improving it? So that git history mark this contribution credited by you.
